### PR TITLE
2015.5 backport tls changes from develop

### PR DESCRIPTION
**Pull request by joejulian**
Merging in the develop bugfixes and the filename feature for inclusion in 2015.5.3

This also simplifies maintenance of tls_test as there's only one version that fits both branches.

### CODE REVIEW

File: salt/modules/tls.py [hunk 1]
```diff
@@ -48,15 +136,32 @@ def __virtual__():
     '''
     Only load this module if the ca config options are set
```
Comment (kiorky) [inline salt/modules/tls.py]: @joejulian this CRITICAL bug needs to be addressed first (#24480)

File: salt/modules/tls.py [hunk 2]
```diff
@@ -22,23 +103,30 @@
...
-import salt.utils
 
 
 log = logging.getLogger(__name__)
 
+two_digit_year_fmt = "%y%m%d%H%M%SZ"
+four_digit_year_fmt = "%Y%m%d%H%M%SZ"
+
 # Always use UTC for certificate info
 os.environ['TZ'] = 'UTC'
```
Comment (kiorky) [inline salt/modules/tls.py]: @joejulian this CRITICAL bug needs to be addressed first (#24480)

**Comment by terminalmage** [comment]
This thoroughly breaks the tls execution module on RHEL 6, as 0.10 is the latest version of pyOpenSSL available.

**Comment by joejulian** [comment]
Thanks, I'll take a look at that asap.

**Comment by terminalmage** [comment]
Thanks, @joejulian!

**Comment by terminalmage** [comment]
@joejulian, we are tracking this in https://github.com/saltstack/salt/issues/25384, please reference that issue number in any related pull requests. Thanks again for taking a look at this.

**Pull request closed without merging**